LEAF_Request_Portal/admin/templates/mod_workflow.tpl

```diff
@@ -2177,85 +2181,89 @@
+                    workflowID = workflow_id;
+                    dialog.hide();
+                });
 
-            workflows[workflowID] = workflows[currentWorkflow];
-            workflows[workflowID]['workflowID'] = parseInt(workflowID);
-            workflows[workflowID]['description'] = title;
-            old_steps[-1] = -1;
+                workflows[workflowID] = workflows[currentWorkflow];
+                workflows[workflowID]['workflowID'] = parseInt(workflowID);
+                workflows[workflowID]['description'] = title;
+                old_steps[-1] = -1;
 
-            for(let i in steps) {
-                // add step, if successful update that step
-                addStep(workflowID, steps[i].stepTitle, function(stepID) {
+                for(let i in steps) {
+                    // add step, if successful update that step
+                    addStep(workflowID, steps[i].stepTitle, function(stepID) {
 
-                    if (isNaN(stepID)) {
-                        console.log(stepID);
-                    } else {
-                        old_steps[steps[i].stepID] = stepID;
-                        updatePosition(workflowID, stepID, steps[i].posX, steps[i].posY);
+                        if (isNaN(stepID)) {
+                            console.log(stepID);
+                        } else {
+                            old_steps[steps[i].stepID] = stepID;
+                            updatePosition(workflowID, stepID, steps[i].posX, steps[i].posY);
 
-                        updateTitle(steps[i].stepTitle, stepID, function(res) {
-                            // Alls well that ends well.
-                        });
+                            updateTitle(steps[i].stepTitle, stepID, function(res) {
+                                // Alls well that ends well.
+                            });
 
-                        if (steps[i].stepData != null) {
-                            let auto = JSON.parse(steps[i].stepData);
+                            if (steps[i].stepData != null) {
+                                let auto = JSON.parse(steps[i].stepData);
 
-                            let seriesData = {
-                                AutomatedEmailReminders: {
-                                    'Automate Email Group': auto.AutomatedEmailReminders.AutomateEmailGroup,
-                                    'Days Selected': auto.AutomatedEmailReminders.DaysSelected,
-                                    'Date Selected': auto.AutomatedEmailReminders?.DateSelected || '',
-                                    'Additional Days Selected': auto.AutomatedEmailReminders.AdditionalDaysSelected,
-                                }
-                            };
+                                let seriesData = {
+                                    AutomatedEmailReminders: {
+                                        'Automate Email Group': auto.AutomatedEmailReminders.AutomateEmailGroup,
```

Comment:
Do we need to think about having "default" values for these?  Or are we sure there will always be values and these assignments will never fail to a null?

Reply:
This part was also unchanged.  I think it should be ok though - if stepData isn't null, the field should have these.  The only one that was in question (because emails by date was added later) was Date Selected, which already has a default value.  If additional kinds of stepData are added in the future I agree we should re-visit this, but I also don't want to mess with email reminder default values without looking at the related functionality in more detail.